xknx/dpt/dpt_hvac_mode.py

```diff
@@ -30,46 +30,70 @@ class HVACOperationMode(Enum):
     NODEM = "NoDem"
 
 
-class DPTHVACMode(DPTBase):
-    """
-    Abstraction for KNX HVAC mode.
+class _DPTClimateMode(DPTBase):
+    """Base class for KNX Climate modes."""
 
-    DPT 20.102
-    """
+    SUPPORTED_MODES = {}
 
     @classmethod
     def from_knx(cls, raw):
         """Parse/deserialize from KNX/IP raw data."""
         cls.test_bytesarray(raw, 1)
-        if raw[0] == 0x04:
-            return HVACOperationMode.FROST_PROTECTION
-        if raw[0] == 0x03:
-            return HVACOperationMode.NIGHT
-        if raw[0] == 0x02:
-            return HVACOperationMode.STANDBY
-        if raw[0] == 0x01:
-            return HVACOperationMode.COMFORT
-        if raw[0] == 0x00:
-            return HVACOperationMode.AUTO
+        if raw[0] in cls.SUPPORTED_MODES:
+            return cls.SUPPORTED_MODES[raw[0]]
         raise CouldNotParseKNXIP("Could not parse HVACOperationMode")
 
     @classmethod
     def to_knx(cls, value):
         """Serialize to KNX/IP raw data."""
-        if value == HVACOperationMode.AUTO:
-            return (0,)
-        if value == HVACOperationMode.COMFORT:
-            return (1,)
-        if value == HVACOperationMode.STANDBY:
-            return (2,)
-        if value == HVACOperationMode.NIGHT:
-            return (3,)
-        if value == HVACOperationMode.FROST_PROTECTION:
-            return (4,)
+        for knx_value, mode in cls.SUPPORTED_MODES.items():
+            if mode == value:
+                return (knx_value,)
         raise ConversionError("Could not parse HVACOperationMode", value=value)
+        # TODO serialize cls.__name__ ??
+
+
+class DPTHVACContrMode(_DPTClimateMode):
+    """
+    Abstraction for KNX HVAC controller mode.
+
+    DPT 20.105
+    """
+
+    SUPPORTED_MODES = {
+        0: HVACOperationMode.AUTO,
+        1: HVACOperationMode.HEAT,
+        2: HVACOperationMode.MORNING_WARMUP,
+        3: HVACOperationMode.COOL,
+        4: HVACOperationMode.NIGHT_PURGE,
+        5: HVACOperationMode.PRECOOL,
+        6: HVACOperationMode.OFF,
+        7: HVACOperationMode.TEST,
+        8: HVACOperationMode.EMERGENCY_HEAT,
+        9: HVACOperationMode.FAN_ONLY,
+        10: HVACOperationMode.FREE_COOL,
+        11: HVACOperationMode.ICE,
+        14: HVACOperationMode.DRY,
+        20: HVACOperationMode.NODEM}
+
+
+class DPTHVACMode(_DPTClimateMode):
+    """
+    Abstraction for KNX HVAC mode.
+
+    DPT 20.102
+    """
+
+    SUPPORTED_MODES = {
+        0: HVACOperationMode.AUTO,
```

Comment:
Wouldn't it make sense to use the official KNX terms here?

Defintion from KNX Standard v2.1:
![image](https://user-images.githubusercontent.com/114137/85108142-7a0ac080-b20f-11ea-99d1-c02bf4af6b67.png)

Reply:
good point. I just copied what was already there. I think we should totally use the knx values and just map it to fitting HA values in the HA integration

Reply:
How about we add a new config option *_eco and mark the existing one as deprecated but keep it around for some time? And also add the option for "standby" which I think is currently missing.

Reply:
Looking at the MDT actuator "Standby" is assumed when the other 3 are set to 0. There is no communication object for "Standby". Are you aware if this the case for other models / are there even other manufacturers using binary mode objects?

Reply:
Yes, my Gira room controller has four separate 1bit channels. See page 54 of the manual, functions 82-85: https://download.gira.de/data3/51423490.pdf

Reply:
Right, I just had a look at my Jung actuator 2336. It also has a standby CO. Not really sure how to implement this without a new configuration mode. Maybe we could just add a note in the documentation to manually set `operation_modes` when using the MDT type.

Btw. Jung, Gira and also MDT refer to eco mode as "Night" in their (german) manuals.